NewPlatform.Flexberry.ORM.ODataService/Expressions/FilterBinder.cs

```diff
@@ -2102,37 +2102,64 @@ private void FillFilterDetailProperties(LambdaNode anyNode)
                 var it = firstNode.Source as ResourceRangeVariableReferenceNode;
                 if (it != null && it.Name == "$it")
                 {
-                    var binaryNode = anyNode.Body as BinaryOperatorNode;
-                    var propertyNodes = new SingleValuePropertyAccessNode[0];
-                    if (binaryNode != null)
+                    GetFilterDetailPropertiesFromNode(anyNode.Body, firstNode.NavigationProperty.Name, FilterDetailProperties);
+                }
+            }
+        }
+
+        private void GetFilterDetailPropertiesFromNode(SingleValueNode node, string navigationPropertyName, List<string> filterDetailProperties)
+        {
+            if (node == null)
+            {
+                return;
+            }
+
+            SingleValuePropertyAccessNode[] propertyNodes = new SingleValuePropertyAccessNode[0];
+            var binaryNode = node as BinaryOperatorNode;
+
+            if (binaryNode != null)
+            {
+                if (binaryNode.Left is SingleValuePropertyAccessNode)
+                {
+                    propertyNodes = new SingleValuePropertyAccessNode[] { binaryNode.Left as SingleValuePropertyAccessNode, binaryNode.Right as SingleValuePropertyAccessNode };
+                }
+                else if (binaryNode.Left is SingleValueFunctionCallNode)
+                {
+                    GetFilterDetailPropertiesFromNode(binaryNode.Left, navigationPropertyName, filterDetailProperties);
+
+                    if (binaryNode.Right is SingleValueFunctionCallNode)
                     {
-                        propertyNodes = new SingleValuePropertyAccessNode[] { binaryNode.Left as SingleValuePropertyAccessNode, binaryNode.Right as SingleValuePropertyAccessNode };
+                        GetFilterDetailPropertiesFromNode(binaryNode.Right, navigationPropertyName, filterDetailProperties);
                     }
+                }
+            }
 
-                    var functionCallNode = anyNode.Body as SingleValueFunctionCallNode;
-                    if (functionCallNode != null && functionCallNode.Name == "contains")
+            var functionCallNode = node as SingleValueFunctionCallNode;
+            if (functionCallNode != null && functionCallNode.Name == "contains")
+            {
+                var parameters = functionCallNode.Parameters.ToList();
```

Comment:
What about this approach?
```suggestion
                propertyNodes = functionCallNode.Parameters.OfType<SingleValuePropertyAccessNode>().ToList();
```